src/http/server.ts

```diff
@@ -556,114 +576,52 @@ export class Server {
-        // optional parameters (:name?), and required parameters before optional params.
-        // This is what we do to construct the `regex_path`. I haven't been able to
-        // come up with a regex that would replace all instances and work, which is
-        // why a loop is being used here, to replace the first instance of an optional
-        // parameter (to account for a possible required parameter before), and then
-        // replace the rest of the occurrences. It's slightly tricky because the path
-        // `/users/:name?/:age?/:city?` should match  `/users`.
-        for (let i = 0; i < maxOptionalParams; i++) {
-          if (i === 0) { // We need to mark the start for the first optional param
-            // The below regex is very similar to `REGEX_URI_MATCHES` but this regex isn't
-            // global, and accounts for there being a required parameter before
-            tmpPath = tmpPath.replace(
-              /\/(:[^(/]+|{[^0-9][^}]*}\?)\/?/,
-              "/?([a-zA-Z0-9]+)?/?", // A `/` being optional, as well as the param being optional, and a ending `/` being optional
-            );
-          } else { // We can now create the replace regex for the rest taking into consideration the above replace regex
-            tmpPath = tmpPath.replace(
-              /\/?(:[^(/]+|{[^0-9][^}]*}\?)\/?/,
-              "([^/]+)?/?",
-            );
-          }
-        }
-        const pathObj = {
-          og_path: path,
-          regex_path: `^${tmpPath}$`,
-          // Regex is same as other blocks, but we also strip the `?`.
-          params: (path.match(Server.REGEX_URI_MATCHES) || []).map(
-            (element: string) => {
-              return element.replace(/:|{|}|\?/g, "");
-            },
-          ),
-        };
-        newPaths.push(pathObj);
+        paths = this.getResourcePathsUsingWildcard(path);
+
+        // Handle optional params
+      } else if (path.includes("?") === true) {
+        paths = this.getResourcePathsUsingOptionalParams(path);
+
+        // Handle basic paths that don't include wild cards or optional params
       } else {
-        const pathObj = {
-          og_path: path,
-          regex_path: `^${
-            path.replace(
-              Server.REGEX_URI_MATCHES,
-              Server.REGEX_URI_REPLACEMENT,
-            )
-          }/?$`,
-          params: (path.match(Server.REGEX_URI_MATCHES) || []).map(
-            (element: string) => {
-              return element.replace(/:|{|}/g, "");
-            },
-          ),
-        };
-        newPaths.push(pathObj);
+        paths = this.getResourcePaths(path);
       }
-    }
 
-    resourceClass.paths_parsed = newPaths;
+      resourceParsedPaths.push(paths);
+
+      // Include the regex path in the index, so we can search for the regex
+      // path during runtime in `.getResourceClass()`
+      this.resource_index_service.addItem(paths.regex_path, resourceClass);
```

Comment:
adding an item should take in an array of items so that search terms can be built in one call